src/map/script-fun.cpp

```diff
@@ -2693,6 +2693,24 @@ void builtin_npctalk(ScriptState *st)
     }
 }
 
+static
+void builtin_walk(ScriptState *st)
+{
+    dumb_ptr<npc_data> nd;
+    short to_x = conv_num(st, &AARG(0));
+    short to_y = conv_num(st, &AARG(1));
+    dumb_ptr<npc_data> nd_ = map_id_is_npc(st->oid);
+    assert (nd_ && nd_->npc_subtype == NpcSubtype::SCRIPT);
+    nd = nd_->is_script();
+
+    nd->to_x = to_x;
+    nd->to_y = to_y;
+    nd->bl_x = to_x;
```

Comment:
This would be only for scripted event walking. Not random movement of NPCs. I see what you got here. Let me work on this a bit.